Add setup.py

tiran opened the pull request:
``venv/bin/pip install .`` installs all dependencies and
venv/bin/cherry_picker script.

Signed-off-by: Christian Heimes <christian@python.org>

Closes https://github.com/python/core-workflow/issues/50
Closes https://github.com/python/core-workflow/issues/64
Closes https://github.com/python/core-workflow/issues/89

Mariatta commented:
Thanks @tiran I believe this will close https://github.com/python/core-workflow/issues/64 So I edited the description.

Will actually review later this evening :)

tiran commented:
@Mariatta I have amended the commit and updated the readme as request. I also replaced ``(venv) $ python -m cherry_picker`` with just ``(venv) $ cherry_picker``.

Mariatta commented:
Thanks @tiran :)
I tested this PR, and made some adjustment to the documentation.

@brettcannon or @ncoghlan: Could one of you review this too please?
I'm trying to simplify the setup process, and making the assumption that most people will prefer to cherry_pick from their existing cloned CPython directory instead of cloning it again. So I updated that part of the readme. Thanks.

ncoghlan commented on setup.py:
```diff
@@ -0,0 +1,33 @@
+    license='Apache',
+    url='https://github.com/python/core-workflow',
+    author='Mariatta Wijaya',
+    author_email='mariatta.wijaya@gmail.com',
```
You may want to put `core-workflow@gmail.com` as your address here, so folks know to contact the list with any questions, rather than contacting you directly.

tiran replied:
I added @Mariatta 's mail address. We may want to keep it and add maintainer / maintainer_email for contact. After all Marietta wrote the script and deserves credits. :)

Mariatta replied:
Thanks 😸  Well I've prepared a canned response already "Please file an issue at python/core-workflow" 

I'll add the following:
```
"maintainer": "Python Core Developers",
"maintainer_email": "core-workflow@mail.python.org",
```

Question: is `version` metadata not needed? The [doc](https://docs.python.org/3.6/distutils/setupscript.html#meta-data) seems to indicate that it's required.

Mariatta commented:
Thanks @tiran
You closed 5 open issues with one PR! :tada: :1st_place_medal:
